### ISSUE TITLE
repeat

### ISSUE THREAD

**Issue by haviduck** [biparnakroy/artificialYOLOtrainer#4]
hey! cool stuff. im currently working on a similar system. what about repeating the process after the new weight is trained? to kinda keep getting better weights?  any thoughts?

**Comment by biparnakroy** [comment]
Yes I had this idea, and what i did is i kept on training the models on the subsequent pre-trained weights and along side that i kept on increasing the size of my dataset as well, so I was generating new dataset as well as training my model.
![image](https://user-images.githubusercontent.com/54219771/132367282-979b7ceb-54c5-4c31-9e8d-1740aa52ab4f.png)

This is a diagrammatic representation of the process i tried to explain. 

In this implementation I had increased the size of the subsequent datasets by 2 times, but this is can be experimented with.

**Comment by haviduck** [comment]
EXACTLY what i was thinking about. so i kinda do this manually now but in my specific case i also wanted to expand the dataser per image for different weather and lighting conditions. so using the img_aug library i do a post process operation where i dupe all the annotated images, make 4 different variarions for night, day, fog, rain.  the original annotations follow making my dataset wider while keeping the accuracy. im still in a research stage of it, particularly night emulation cause shadows are tough, but i really want to see what we can do. i feel pseudolabelling is so friggin close to being nonsupervised and it would elevate one of the shittiest jobs in cv to a breeze.   1 perfect starting model is all it takes.. so cool.
